src/PKSim.Presentation/Services/RenameBuildingBlockTask.cs

```diff
@@ -241,16 +241,15 @@ private FormulaUsablePath formulaUsablePathReferencing(IFormula formula, IFormul
 
       private IEnumerable<IContainer> getContainersToRename(Simulation simulation, IPKSimBuildingBlock templateBuildingBlock, string oldContainerName)
       {
+         var containersToRename = new List<IContainer>();
+
          var rootContainer = simulation.Model.Root.EntityAt<IContainer>(getContainerPathToRename(templateBuildingBlock.BuildingBlockType));
          if (templateBuildingBlock.BuildingBlockType != PKSimBuildingBlockType.Formulation)
-            yield return rootContainer.Container(oldContainerName);
+            containersToRename.Add(rootContainer.Container(oldContainerName));
          else
-         {
-            foreach (var formulationContainer in rootContainer.GetAllChildren<IContainer>(x => x.IsNamed(oldContainerName) && x.ContainerType == ContainerType.Formulation))
-            {
-               yield return formulationContainer;
-            }
-         }
+            return rootContainer.GetAllChildren<IContainer>(x => x.IsNamed(oldContainerName) && x.ContainerType == ContainerType.Formulation);
+         
+         return containersToRename.Where(x => x != null);
```

Comment:
still looks a bit weird since this is only use in one if but created globally. 
What about you just return new []{rootContainer.Container(oldContainerName)}.where(x=>x!=null)
and put a comment above explaining why you are filtering for null? This is unclear to me that this is because stuff might have been gone missing between project version

Reply:
I'll need you to write that comment. I am not familiar with the change that caused it. I think you said that

Reply:
Ok I'll push to this PR